components/eam/src/chemistry/modal_aero/modal_aero_coag.F90

```diff
@@ -1,2337 +1,1363 @@
 
-! *** set up for second moment
-!      the second moment equations are new and begin with equations a1
-!     through a4 of binkowski and shankar (1995). after some algebraic
-!     rearrangement and application of the extended mean value theorem
-!     of integral calculus, equations are obtained that can be solved
-!     analytically with correction factors as has been done by
-!     whitby et al. (1991)
+      call qnum_update_self_and_intermodal_coag( rateij, rateii, qnum_bgn(nait), qnum_end(nait) )
 
-! *** the term ( dp1 + dp2 ) ** (2/3) in equations a3 and a4 of
-!     binkowski and shankar (1995) is approximated by
-!     (dgat ** 3 + dgac **3 ) ** 2/3
+      qnum_tavg(nait) = (qnum_bgn(nait) + qnum_end(nait))*0.5_wp
 
-! *** near-continuum form
 
-      i1nc = knc * dgat2 * (   &
-             two * esat16   &
-           + r2 * esat04 * esac04   &
-           + ri2 * esat36 * esac04   &
-           + a * kngat * (   &
-                 esat04   &
-           +     ri2 * esat16 * esac04   &
-           +     ri4 * esat36 * esac16   &
-           +     r2 * esac04 )  )
+end subroutine mam_coag_num_update
 
 
+subroutine qnum_update_selfcoag( ybetajj0, deltat, qnum_bgn, qnum_end )
+!-----------------------------------------------------------------------------------------
+! Purpose: update the number mixing ratio of a single mode by considering self-coagulation
+!-----------------------------------------------------------------------------------------
 
+    real(wp),intent(in)  :: qnum_bgn   ! qnum (number mixing ratio) start value
+    real(wp),intent(in)  :: ybetajj0   ! self-coagulation coefficient
+    real(wp),intent(in)  :: deltat     ! timestep length
+    real(wp),intent(out) :: qnum_end   ! qnum (number mixing ratio) end value
 
-! *** free-molecular form
+    ! Analytical solution
+    qnum_end = qnum_bgn / ( 1.0_wp + ybetajj0*deltat*qnum_bgn )
 
-       i1fm =  kfmatac * sqdgat5 * bm2ij(n1,n2n,n2a) * (   &
-               esat25   &
-            +  two * r2 * esat09 * esac04   &
-            +  rx4 * esat01 * esac16   &
-            +  ri3 * esat64 * esac09   &
-            +  two * ri1 * esat36 * esac01   &
-            +  r * esat16 * esac01  )
+end subroutine qnum_update_selfcoag
 
+subroutine qnum_update_self_and_intermodal_coag( rateij, rateii, qnum_bgn, qnum_end )
+!-----------------------------------------------------------------------------------------
+! Purpose: update the number mixing ratio of a single mode considering self-coagulation 
+!          and inter-modal coagulation
+!-----------------------------------------------------------------------------------------
 
+    real(wp),intent(in)  :: rateij, rateii  ! inter-modal and self-coagulation rates
+    real(wp),intent(in)  :: qnum_bgn        ! qnum (number mixing ratio) start value
+    real(wp),intent(out) :: qnum_end        ! qnum (number mixing ratio) end value
 
-! *** loss to accumulation mode
+    real(wp) :: tmpc
```

Comment:
Can we rename `tmpc` to `exp_rate` or `tmp_exp` or something similar?

Reply:
Changed to `tmp_exp` in [this commit](https://github.com/eagles-project/e3sm_mam4_refactor/pull/32/commits/0aacb82f99405c7f4207680de523988fb31ce395).